### ISSUE TITLE
Feature: Hot module replacement

### ISSUE THREAD

**Issue by pantharshit00** [abelljs/abell#60]
Just persisting an idea for the future. I know it is a lot of work.

**Is your feature request related to a problem? Please describe.**
Right now the dev server reloads the whole app in order to show the updated content. Most popular static generators or apps use hot module replacement to provide a much smoother experience

**Describe the solution you'd like**
Implement a hot module replacement strategy for abell so that it can provider a richer dev experience

**Additional context**
React hot loader(https://github.com/gaearon/react-hot-loader) (note: deprecated in favour of fast refresh)
webpack hot loader: https://webpack.js.org/concepts/hot-module-replacement/

**Comment by saurabhdaware** [comment]
I in fact implemented no refresh content change in one of the initial versions but the problem I faced was, if I replace particular content, I also have to execute scripts related to it. So we somehow need reference of which scripts are used by that particular html element.

For example, when I tried to replace content, it did work but then the code-highlighting stopped working because it didn't execute the highlighting init script so we somehow need to tackle that.
